src/robot/client.ts

```diff
@@ -370,6 +370,7 @@ export class RobotClient extends EventDispatcher implements Robot {
   }
 
   public async connect(
+    priority?: number,
     authEntity = this.savedAuthEntity,
     creds = this.savedCreds
   ) {
```

Comment:
Since we are updating the arguments here, could we update `connect` to accept an `object` of options? Should making updating this signature (especially with optional properties) much safer/easier moving forward.

```suggestion
  export interface ConnectOptions {
    priority?: number;
    authEntity?: string;
    creds?: Credentials; 
  }

  public async connect({
    priority,
    authEntity = this.savedAuthEntity,
    creds = this.savedCreds,
  }: ConnectOptions) {
```

cc: @micheal-parks

Reply:
I actually really like this idea. Thank you for bringing it up! I noticed this as well and wished I could do this, but wasn't sure if it was a good idea to change such an important function. I'll be making this change now that I know others agree with me!